Automattic/wp-calypso
# Writing prompts: fix Home card analytics issues

#### Proposed Changes

<img width="605" alt="image" src="https://user-images.githubusercontent.com/1699996/217071536-2c6288e6-560a-4049-994e-7a3a1a361564.png">

Fixes these issues with the My Home writing prompts card

- Fixes a bug where `calypso_customer_home_answer_prompt` and `calypso_customer_home_skip_prompt` events are triggered on page load, rather than when clicking to answer a prompt.
- Prevents JS errors if a malformed API response returns an empty prompt.
- Prevents a race condition where the page may navigate away before logging events when answering a prompt
- React development errors for `stroke-width` attribute and missing key in `answered_users_sample` list.

#### Testing Instructions

* Using a blog that is set to show posts on the home page, visit `/home/{site}` and find the Daily writing prompt card.
* In the JS console of the browser, use `localStorage.setItem( 'debug', 'calypso:analytics*' );` to enable tracks debugging, and reload the page.
* Make sure the `calypso_customer_home_answer_prompt` and `calypso_customer_home_skip_prompt` events **do not** trigger when the page is loaded.
* Click on "View all responses". You should see the `calypso_customer_home_view_all_responses` event logged in the console and navigate to the reader tag page for the prompt.
* Go back and click the "Post Answer" button. You should see the `calypso_customer_home_answer_prompt` event logged and navigate to the editor.
* Go back, change the prompt, and click on the "Post Answer" button again. You should additionally see the `calypso_customer_home_skip_prompt` even recorded.
* Switch to an Atomic site that loads the editor outside calypso, and test that the Post Answer button works in the same way.
